extensions/guacamole-auth-nextcloud/src/main/java/org/apache/guacamole/auth/nextcloud/NextcloudJwtAuthenticationProvider.java

```diff
@@ -0,0 +1,249 @@
+            }
+            logger.info("Token valid.");
+        } catch (final GuacamoleException ex) {
+            logger.error("Token validation failed.", ex);
+            throw new GuacamoleException(ex.getMessage());
+        }
+        return null;
+
+    }
+
+    /**
+     * Validates the provided JSON Web Token (JWT).
+     *
+     * This method decodes the public key from a base64 encoded string, verifies the JWT using
+     * the ECDSA256 algorithm, and checks the token's validity period and user permissions.
+     *
+     * @param token
+     *     The JWT token to validate.
+     * @return {@code true}
+     *     If the token is valid and the user is allowed, {@code false} otherwise.
+     * @throws GuacamoleException
+     *     If the user is not allowed or the token is expired.
+     * @throws JWTVerificationException
+     *     If the token verification fails.
+     * @throws NoSuchAlgorithmException
+     *     If the algorithm for key generation is not available.
+     * @throws InvalidKeySpecException
+     *     If the key specification is invalid.
+     */
+    private boolean isValidJWT(final String token) throws GuacamoleException {
+        try {
+            // Decode the Base64 encoded public key from configuration and generate the ECPublicKey object.
+            byte[] keyBytes = Base64.getDecoder().decode(confService.getPublicKey());
+            KeyFactory keyFactory = KeyFactory.getInstance("EC");
+            X509EncodedKeySpec keySpec = new X509EncodedKeySpec(keyBytes);
+            ECPublicKey publicKey = (ECPublicKey) keyFactory.generatePublic(keySpec);
+
+            // Create a JWT verifier instance with the provided public key, verify the token and decode the content.
+            JWTVerifier verifier = JWT.require(Algorithm.ECDSA256(publicKey)).build();
+            DecodedJWT decodedJWT = verifier.verify(token);
+
+            // Check if the user extracted from the token's payload is allowed to open Guacamole login page.
+            boolean isUserAllowed = this.isUserAllowed(decodedJWT.getPayload());
+            if (!isUserAllowed) {
+                throw new GuacamoleException("User not allowed.");
+            }
+
+            // Validate the token's expiration by comparing the current date with the token's expiration date,
+            // ensuring it falls within the acceptable validity duration defined by MINUTES_TOKEN_VALID.
+            Date currentDate = new Date();
+            Date maxValidDate = new Date(currentDate.getTime() - (MINUTES_TOKEN_VALID * 60 * 1000));
+            boolean isValidToken = decodedJWT.getExpiresAt().after(maxValidDate);
+            if (!isValidToken) {
+                throw new GuacamoleException("Token expired.");
+            }
+
+            return true;
+        } catch (final JWTVerificationException | NoSuchAlgorithmException | InvalidKeySpecException ex) {
+            logger.error("Token validation failed.", ex);
+            return false;
+        }
```

Comment:
* Please don't cuddle the braces.
* Should this `return false`, or should it re-throw these exceptions as a `GuacamoleSecurityException`?

Reply:
Oh yes, `GuacamoleSecurityException` is much better. Thanks for the hint :+1: